scripts/genie.lua

```diff
@@ -366,14 +366,21 @@ function exampleProjectDefaults()
 	configuration { "android*" }
 		kind "ConsoleApp"
 		targetextension ".so"
-		linkoptions {
-			"-shared",
-		}
 		links {
 			"EGL",
 			"GLESv2",
 		}
 
+	configuration { "android*", "Debug" }
+		linkoptions {
+			"-shared -Wl,-soname,lib" .. project().name .. "Debug.so"
```

Comment:
You should keep this separate.
Leave `linkoptions { "-shared" }` above, and add these new options without it below.